diff --git a/jam-files/xmlrpc-c.jam b/jam-files/xmlrpc-c.jam
index 0a51f4f..1c04321 100644
--- a/jam-files/xmlrpc-c.jam
+++ b/jam-files/xmlrpc-c.jam
@@ -78,18 +78,19 @@ else
     }
 
     # now add stuff to the requirements
-    local xmlrpc-cxxflags = [ shell_or_die "$(config) c++2 abyss-server --cflags" ] ;
+    local xmlrpc-cxxflags = [ shell_or_die "$(config) client c++2 abyss-server --cflags" ] ;
     requirements += <define>HAVE_XMLRPC_C ;
     requirements += <cxxflags>$(xmlrpc-cxxflags) ;
     
-    local libs = [ shell_or_die "$(config) c++2 abyss-server --libs" ] ;
+    # local libs = [ shell_or_die "$(config) client c++2 abyss-server --libs" ]; 
+    # libs[0] += " -lxmlrpc_client++" ;
+    libs += "-L/usr/lib/x86_64-linux-gnu -L/home/alex/mosesdecoder/opt/lib   -lxmlrpc_server_abyss++ -lxmlrpc_server++ -lxmlrpc_abyss++ -lxmlrpc_server_abyss -lxmlrpc_server -lxmlrpc_abyss  -lxmlrpc++ -lxmlrpc_util++ -lxmlrpc_client -lxmlrpc -lxmlrpc_xmlparse -lxmlrpc_xmltok -lxmlrpc_util -lpthread -L/usr/lib/x86_64-linux-gnu -lxmlrpc_client++ -lidn -lrtmp -lssl -lcrypto -lssl -lcrypto -llber -lldap -lz " ;
     for local i in [ SPLIT_BY_CHARACTERS $(libs) : " " ] 
     { 
       local libname = [ MATCH "-l(xmlrpc.*)" : $(i) ] ; 
       if $(libname) 
       {   
 	external-lib $(libname) : $(prefix)/lib ;
-	#      : : <runtime-link>static:<link>static <runtime-link>shared:<link>shared ;
 	requirements += <library>$(libname) ;
       }
       local pathname = [ MATCH "-L(.*)" : $(i) ] ; 
@@ -113,8 +114,30 @@ else
   {
     rule xmlrpc ( what ? ) { } # never return anything
   }
-  
-}
 
 
+## die in a fire. there's got to be a better way to make this work.
+lib idn : : <name>idn <search>/usr/lib/x86_64-linux-gnu ;
+lib rtmp : : <name>rtmp <search>/usr/lib/x86_64-linux-gnu ;
+lib ssl : : <name>ssl <search>/usr/lib/x86_64-linux-gnu ;
+lib crypto : : <name>crypto <search>/usr/lib/x86_64-linux-gnu ;
+lib lber : : <name>lber <search>/usr/lib/x86_64-linux-gnu ;
+lib ldap : : <name>ldap <search>/usr/lib/x86_64-linux-gnu ;
+lib curl : : <name>curl <search>/usr/lib/x86_64-linux-gnu ;
+lib gnutls : : <name>gnutls <search>/usr/lib/x86_64-linux-gnu ;
+lib gcrypt : : <name>gcrypt <search>/usr/lib/x86_64-linux-gnu ;
+lib tasn1 : : <name>tasn1 <search>/usr/lib/x86_64-linux-gnu ;
 
+requirements += <library>idn ;
+requirements += <library>rtmp ;
+requirements += <library>ssl ;
+requirements += <library>crypto ;
+requirements += <library>lber ;
+requirements += <library>ldap ;
+requirements += <library>curl ;
+requirements += <library>gnutls ;
+requirements += <library>gcrypt ;
+requirements += <library>z ;
+requirements += <library>tasn1 ;
+  
+}
